openstack/baremetal/v1/ports/requests.go

```diff
@@ -0,0 +1,223 @@
+package ports
+
+import (
+  "fmt"
+
+  "github.com/gophercloud/gophercloud"
+  "github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to the
+// List request.
+type ListOptsBuilder interface {
+	ToPortListQuery() (string, error)
+  ToPortListDetailQuery() (string, error)
```

Comment:
Indentation is off here and elsewhere, I would run `gofmt` on all your files. go is rather opinionated, there's "one" way to format the code, I use this in vim: https://github.com/stbenjam/dotfiles/blob/master/vimrc#L73.  GoLand IDE also does it.  There's git hooks out there too to run the CLI before commit.